### PR TITLE
Beam() now pools, tesla engine sound effects

### PR DESCRIPTION
**Pull request by Iamgoofball**
see title

sound effects are the ones we use for LIGHTNING BOLT already

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
I'd recommend profiling pooling, first, before doing this.

Reason being....even with the pooling optimizations, I strongly suspect pooling still incurs a loss of performance over `qdel`.

I tested it quite thoroughly, and even if a number of the pooling procs were literally *costless*, qdel'ing them was still cheaper.

**Comment by MrStonedOne** [comment]
there is a bit of overhead that isn't profiled.

the cost of deallocating objects.

recently i've found this to be about the amount of 8 proc calls.

File: code/datums/beam.dm [hunk 1]
```diff
@@ -54,7 +54,7 @@
 
 /datum/beam/proc/Reset()
 	for(var/obj/effect/ebeam/B in elements)
-		qdel(B)
+		PlaceInPool(B)
```
Comment (duncathan) [inline code/datums/beam.dm]: No jackass, keep the qdels
Make ebeam/Destroy() return QDEL_HINT_PUTINPOOL

Comment (Iamgoofball) [inline code/datums/beam.dm]: I thought this is what I was supposed to do, sorry

File: code/datums/beam.dm [hunk 2]
```diff
@@ -45,7 +45,7 @@
 			Draw()
 		sleep(sleep_time)
 
-	qdel(src)
+	PlaceInPool(src)
```
Comment (Fox-McCloud) [inline code/datums/beam.dm]: First off, as @duncathan pointed out, this isn't the proper way to do it....secondly..

You're pooling the beam datum, but yet never calling PoolOrNew() for the beam datum anywhere; it's senseless to let these accumulate in a pool, but never get pulled back out.

**Pull request closed without merging**